### PR TITLE
Remasc reworked

### PR DESCRIPTION
**Pull request by lsebrie**
Changes are introduced to Remasc to enhance performance and reduce disk usage

- now there is a minimum limit for payments, below the value the fees are not distributed
- block information to process remasc is now obtained from blockStore instead of remasc storage
- new tests added to check this behaviors
- values and names for rskip will be changed before final review

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 21 issues
* ![CRITICAL][CRITICAL] 8 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L31](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L31): Unused import - org.apache.commons.lang3.tuple.Pair. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [Bridge.java#L464](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L464): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![CRITICAL][CRITICAL] [Program.java#L187](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L187): Explicitly declare the visibility for "isLogEnabled". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L188](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L188): Explicitly declare the visibility for "isGasLogEnabled". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L229](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L229): Remove this call from a constructor to the overridable "precompile" method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [Program.java#L813](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L813): Method org.ethereum.vm.program.Program.callToAddress(MessageCall) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![CRITICAL][CRITICAL] [Program.java#L1437](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1437): Explicitly declare the visibility for "code". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L1438](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1438): Explicitly declare the visibility for "pc". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![MAJOR][MAJOR] [Bridge.java#L472](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L472): Add the missing @deprecated Javadoc tag. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [Remasc.java#L94](https://github.com/rsksmart/rskj/blob/8056a110457157a643f871d278bcd8b1787268c8/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L94): The Cyclomatic Complexity of this method "processMinersFees" is 14 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 20 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L463](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L463): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![CRITICAL][CRITICAL] [Program.java#L187](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L187): Explicitly declare the visibility for "isLogEnabled". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L188](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L188): Explicitly declare the visibility for "isGasLogEnabled". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L229](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L229): Remove this call from a constructor to the overridable "precompile" method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [Program.java#L813](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L813): Method org.ethereum.vm.program.Program.callToAddress(MessageCall) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![CRITICAL][CRITICAL] [Program.java#L1437](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1437): Explicitly declare the visibility for "code". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L1438](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1438): Explicitly declare the visibility for "pc". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![MAJOR][MAJOR] [Bridge.java#L471](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L471): Add the missing @deprecated Javadoc tag. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [Remasc.java#L94](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L94): The Cyclomatic Complexity of this method "processMinersFees" is 14 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Remasc.java#L128](https://github.com/rsksmart/rskj/blob/bc8a7b6e2dafe7b838fb4a79a6ab5c9c87c367b0/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L128): Method co.rsk.remasc.Remasc.processMinersFees() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 19 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L463](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L463): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![CRITICAL][CRITICAL] [Program.java#L187](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L187): Explicitly declare the visibility for "isLogEnabled". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L188](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L188): Explicitly declare the visibility for "isGasLogEnabled". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L229](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L229): Remove this call from a constructor to the overridable "precompile" method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [Program.java#L813](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L813): Method org.ethereum.vm.program.Program.callToAddress(MessageCall) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![CRITICAL][CRITICAL] [Program.java#L1437](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1437): Explicitly declare the visibility for "code". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![CRITICAL][CRITICAL] [Program.java#L1438](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1438): Explicitly declare the visibility for "pc". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2039)
1. ![MAJOR][MAJOR] [Bridge.java#L471](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L471): Add the missing @deprecated Javadoc tag. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [Remasc.java#L94](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L94): The Cyclomatic Complexity of this method "processMinersFees" is 14 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Remasc.java#L191](https://github.com/rsksmart/rskj/blob/1e3b88144b8baf7eb4956c91e1ca057c7ce7ead3/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L191): Method co.rsk.remasc.Remasc.processMinersFees() accesses list or array with constant index [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ACLI_CONSTANT_LIST_INDEX)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 11 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L454](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L454): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L462](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L462): Add the missing @deprecated Javadoc tag. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [Remasc.java#L94](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L94): The Cyclomatic Complexity of this method "processMinersFees" is 14 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Remasc.java#L191](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L191): Method co.rsk.remasc.Remasc.processMinersFees() accesses list or array with constant index [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ACLI_CONSTANT_LIST_INDEX)
1. ![MINOR][MINOR] [Bridge.java#L491](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L491): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![INFO][INFO] [Bridge.java#L462](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L462): Do not forget to remove this deprecated code someday. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [Remasc.java#L166](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L166): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [Remasc.java#L170](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L170): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [Remasc.java#L245](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L245): Redundant nullcheck of uncles, which is known to be non-null in co.rsk.remasc.Remasc.addNewSiblings() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)
1. ![INFO][INFO] [MainNetFirstForkConfig.java#L3](https://github.com/rsksmart/rskj/blob/1fbf04b0f1ce4704cfe05d38dc593015f1b2747e/rskj-core/src/main/java/org/ethereum/config/blockchain/mainnet/MainNetFirstForkConfig.java#L3): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/remasc/Remasc.java [hunk 1]
```diff
@@ -121,15 +146,22 @@ void processMinersFees() throws IOException, BlockStoreException {
...
         }
 
+        Coin minimumPayableGas = configForBlock.getConstants().getMinimumPayableGas();
         // Takes from rewardBalance this block's height reward.
-        Coin fullBlockReward = rewardBalance.divide(BigInteger.valueOf(remascConstants.getSyntheticSpan()));
-        rewardBalance = rewardBalance.subtract(fullBlockReward);
+        Coin syntheticReward = rewardBalance.divide(BigInteger.valueOf(remascConstants.getSyntheticSpan()));
+        Coin minPayableFees = minimumPayableGas.multiply(executionBlock.getMinimumGasPrice().asBigInteger());
+        if (isRskIp15BisEnabled && syntheticReward.compareTo(minPayableFees) < 0) {
+            logger.debug("Synthetic Reward: {} is lower than minPayableFees: {} at block: {}",
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/remasc/Remasc.java]: `TRACE`?

File: rskj-core/src/main/java/co/rsk/remasc/Remasc.java [hunk 2]
```diff
@@ -170,30 +203,32 @@ void processMinersFees() throws IOException, BlockStoreException {
...
+                syntheticReward = syntheticReward.subtract(punishment);
                 provider.setBurnedBalance(provider.getBurnedBalance().add(punishment));
             }
-            feesPayer.payMiningFees(processingBlockHeader.getHash().getBytes(), fullBlockReward, processingBlockHeader.getCoinbase(), logs);
+            feesPayer.payMiningFees(processingBlockHeader.getHash().getBytes(), syntheticReward, processingBlockHeader.getCoinbase(), logs);
             provider.setBrokenSelectionRule(Boolean.FALSE);
         }
 
-        this.removeUsedSiblings(processingBlockHeader);
+        if (!config.getBlockchainConfig().getConfigForBlock(blockNbr).isRskIp15Bis()) {
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/remasc/Remasc.java]: use `isRskIp15BisEnabled`

File: rskj-core/src/main/java/org/ethereum/config/Constants.java [hunk 3]
```diff
@@ -162,4 +165,8 @@ public int getNewBlockMaxSecondsInTheFuture() {
      * EIP155: https://github.com/ethereum/EIPs/issues/155
      */
     public byte getChainId() { return Constants.CHAIN_ID; }
+
+    public Coin getMinimumPayableGas() {
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/config/Constants.java]: This should go in `RemascConfig`

File: rskj-core/src/main/java/org/ethereum/config/blockchain/DevNetConfig.java [hunk 4]
```diff
@@ -69,4 +69,9 @@ public boolean isRfs94() {
     public boolean isRcs230() {
         return true;
     }
+
+    @Override
+    public boolean isRskIp15Bis() {
+        return true;
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/config/blockchain/DevNetConfig.java]: Shouldn't this be `false`? Deploying this code would make this immediately apply the network upgrade code.

Same for other network configurations.

**Comment by rskops** [comment]
SonarQube analysis reported 11 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L454](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L454): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L462](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L462): Add the missing @deprecated Javadoc tag. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [Remasc.java#L94](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L94): The Cyclomatic Complexity of this method "processMinersFees" is 14 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Remasc.java#L195](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L195): Method co.rsk.remasc.Remasc.processMinersFees() accesses list or array with constant index [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ACLI_CONSTANT_LIST_INDEX)
1. ![MINOR][MINOR] [Bridge.java#L491](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L491): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![INFO][INFO] [Bridge.java#L462](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L462): Do not forget to remove this deprecated code someday. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [Remasc.java#L170](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L170): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [Remasc.java#L174](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L174): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [Remasc.java#L249](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L249): Redundant nullcheck of uncles, which is known to be non-null in co.rsk.remasc.Remasc.addNewSiblings() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)
1. ![INFO][INFO] [MainNetFirstForkConfig.java#L3](https://github.com/rsksmart/rskj/blob/ab0d8ad4959d2532805ea8f45702aead53a61bc6/rskj-core/src/main/java/org/ethereum/config/blockchain/mainnet/MainNetFirstForkConfig.java#L3): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 12 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L454](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L454): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L462](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L462): Add the missing @deprecated Javadoc tag. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
1. ![MAJOR][MAJOR] [Remasc.java#L94](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L94): The Cyclomatic Complexity of this method "processMinersFees" is 15 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Remasc.java#L195](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L195): Method co.rsk.remasc.Remasc.processMinersFees() accesses list or array with constant index [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ACLI_CONSTANT_LIST_INDEX)
1. ![MINOR][MINOR] [Bridge.java#L295](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L295): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MINOR][MINOR] [Bridge.java#L491](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L491): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![INFO][INFO] [Bridge.java#L462](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L462): Do not forget to remove this deprecated code someday. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [Remasc.java#L170](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L170): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [Remasc.java#L174](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L174): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [Remasc.java#L253](https://github.com/rsksmart/rskj/blob/1b5b0c7dce1e29ebd90cff45ede912471ae9d6c7/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L253): Redundant nullcheck of uncles, which is known to be non-null in co.rsk.remasc.Remasc.addNewSiblings() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by aeidelman** [comment]
This PR is replaced by #611. Closing.

**Pull request closed without merging**